### PR TITLE
Docker hardening: network check

### PR DESCRIPTION
**Pull request by glicht**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-4716)

## Description
Add network check to the docker hardening check script

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck.py [hunk 1]
```diff
@@ -150,9 +159,50 @@ def check_cpus(num_cpus: int) -> str:
     return ""
 
 
+def get_default_gateway():
+    res = subprocess.check_output(["ip", "route", "list"], text=True, stderr=subprocess.STDOUT)
+    LOG(f'result of ip route list: {res}')
+    line1 = res.splitlines()[0]
+    if not line1.startswith('default via'):
```
Comment (tkatzir) [inline Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck.py:166]: Maybe protect against case discrepancies?
```suggestion
    if not line1.lower().startswith('default via'):
```

File: Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck.py [hunk 2]
```diff
@@ -150,9 +159,50 @@ def check_cpus(num_cpus: int) -> str:
...
+    return line1.split()[2]
+
+
+def check_network(network_check: str) -> str:
+    """
+    Check that Cloud provider metadata service is not exposed and that access to localhost is not available.
+    """
+    return_res = ""
+    if network_check in ("all", "cloud_metadata"):
+        LOG('Check cloud metadata server access...')
```
Comment (tkatzir) [inline Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck.py:177]: If we're logging, then we can write why we got here ("all" or "cloud_metadata").
```suggestion
        LOG('Check cloud metadata server access... {network_check=}')
```

File: Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck.py [hunk 3]
```diff
@@ -150,9 +159,50 @@ def check_cpus(num_cpus: int) -> str:
...
+        try:
+            res = requests.get(CLOUD_METADATA_URL, timeout=1)
+            LOG(f'cloud metadata server returned successfuly: {res.status_code} {res.headers}')
+            return_res += (f"Access to cloud metadata server: {CLOUD_METADATA_URL} is open. It seems that you haven't blocked "
+                           f"access to the cloud metadata server. Response status code: [{res.status_code}]."
```
Comment (tkatzir) [inline Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck.py:182]:
```suggestion
                           f"access to the cloud metadata server. Response status code: [{res.status_code}]. "
```

File: Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck.py [hunk 4]
```diff
@@ -150,9 +159,50 @@ def check_cpus(num_cpus: int) -> str:
...
+                           f"Response headers: {res.headers}")
+        except Exception as ex:
+            LOG(f'cloud metadata server returned exception (this is good. means there is no access to the server.): {ex}')
```
Comment (tkatzir) [inline Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck.py:185]:
```suggestion
            LOG(f'Cloud metadata server returned an exception (this is good. It means there is no access to the server.): {ex}')
```

File: Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck.py [hunk 5]
```diff
@@ -150,9 +159,50 @@ def check_cpus(num_cpus: int) -> str:
...
+    if network_check in ("all", "host_machine"):
+        LOG("Check host access")
```
Comment (tkatzir) [inline Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck.py:187]:
```suggestion
        LOG("Check host access, {network_check =}"")
```

File: Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck.py [hunk 6]
```diff
@@ -150,9 +159,50 @@ def check_cpus(num_cpus: int) -> str:
...
+        gateway_ip = get_default_gateway()
+        try:
+            res = requests.get(f'https://{gateway_ip}/', verify=False, timeout=1)  # nosec # guardrails-disable-line
+            LOG(f'Host https request returned successfully: {res.status_code} {res.headers}')
+            if return_res:
+                return_res += "\n"
+            return_res += (f"Access to host server via default gateway ip: {gateway_ip} is open. It seems that "
+                           f"you haven't blocked access to the host server. Response status code: [{res.status_code}]."
```
Comment (tkatzir) [inline Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck.py:195]:
```suggestion
                           f"you haven't blocked access to the host server. Response status code: [{res.status_code}]. "
```

File: Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck.py [hunk 7]
```diff
@@ -150,9 +159,50 @@ def check_cpus(num_cpus: int) -> str:
...
+                           f"Response headers: {res.headers}")
+        except Exception as ex:
+            LOG(f'host gateway server returned exception (this is good. means there is no access to the host server.): {ex}')
```
Comment (tkatzir) [inline Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck.py:198]:
```suggestion
            LOG(f'The host gateway server returned an exception (this is good. It means that there is no access to the host server.): {ex}')
```

File: Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck.py [hunk 8]
```diff
@@ -193,7 +247,7 @@ def main():
     return_outputs(table)
     if failed:
         return_error(f'Failed verifying docker hardening:\n{failed_msg}'
-                     'More details at: https://docs.paloaltonetworks.com/cortex/cortex-xsoar/6-0/cortex-xsoar-admin/docker/docker-hardening-guide.html')  # noqa
+                     'More details at: https://docs.paloaltonetworks.com/cortex/cortex-xsoar/6-9/cortex-xsoar-admin/docker/docker-hardening-guide.html')  # noqa
```
Comment (tkatzir) [inline Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck.py:250]: Isn't there a static "latest" address?

**Review by tkatzir — APPROVED** [review]
Great!
Approved, with some minor comments.

File: Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck.py (hunk 8 above)
Comment (glicht) [reply Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck.py:250]: Unfortunately no

File: Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck.py (hunk 5 above)
Comment (glicht) [reply Packs/CommonScripts/Scripts/DockerHardeningCheck/DockerHardeningCheck.py:187]: same as above

**Pull request merged**